src/overdub.jl

```diff
@@ -180,6 +180,9 @@ function overdub_pass!(reflection::Reflection,
     for i in 1:n_method_args
         slot = i + n_prepended_slots
         offset = i + invoke_offset
+        if is_invoke && i == 1
+            invoke_offset += 1
+        end
         if is_invoke && is_calloverload && i == 1
             offset = 2 # 1 is invoke, 2 is f, 3 is Tuple{}, 4... is args
```

Comment:
I don't understand this part of the code, so please check carefully.

Reply:
This does make sense to me. I am not sure about it either.

@jrevels any recollection why it was the way it is?

Reply:
FYI, with this comment I was referring specifically to lines 186-187; I didn't understand why `is_calloverload` is treated separately.

Reply:
Ah that makes sense. I am to blame for that code (https://github.com/jrevels/Cassette.jl/pull/128) 

`is_calloverload` was trying to fix a very similar issue, where when you do call overloading you end up with code that looks similar to what you seem to run into.